spot_rl_experiments/spot_rl/envs/nav_env.py

```diff
@@ -187,16 +190,35 @@ def execute(self, nav_targets_list) -> List[List[Dict]]:
 class SpotNavEnv(SpotBaseEnv):
     def __init__(self, config, spot: Spot):
         super().__init__(config, spot)
-        self.goal_xy = None
+        self._goal_xy = None
+        self._enable_nav_goal_change = False
+        self.backup_get_observation_by_base_fn = self.get_nav_observation
         self.goal_heading = None
         self.succ_distance = config.SUCCESS_DISTANCE
         self.succ_angle = np.deg2rad(config.SUCCESS_ANGLE_DIST)
 
+    def enable_nav_goal_change(self):
+        if not self._enable_nav_goal_change:
+            self._enable_nav_goal_change = True
+            print(
+                f"{self.node_name} Enabling nav goal change get_nav_observation by base switched to get_nav_observation by hand fn"
+            )
+            self.backup_get_observation_by_base_fn = self.get_nav_observation
```

Comment:
@tusharsangam What does base_fn mean?

Reply:
@KavitShah1998 function name is get_nav_observation by default it takes the nav observation by robot's base, we take backup of that function. Thus base_fn means robot's base. To avoid confusion I have changed the function pointer's name to backup_get_nav_observation_by_robot_base